Vic2ToHoI4/Source/HOI4World/States/HoI4States.cpp

```diff
@@ -510,15 +516,63 @@ std::vector<std::set<int>> HoI4::States::consolidateProvinceSets(std::vector<std
 }
 
 
-void HoI4::States::addProvincesAndCoresToNewState(State& newState, const std::set<int>& provinces)
+void HoI4::States::addProvincesAndCoresToNewState(State& newState,
+	 const std::map<std::string, std::unique_ptr<Vic2::Country>>& sourceCountries,
+	 const std::set<int>& provinceNumbers,
+	 const mappers::ProvinceMapper& provinceMapper,
+	 const std::map<int, std::shared_ptr<Vic2::Province>>& vic2Provinces)
 {
-	for (auto province: provinces)
+	std::set<std::pair<std::string, std::string>> possibleCores;
+	for (auto province: provinceNumbers)
 	{
 		newState.addProvince(province);
 		provinceToStateIDMap.insert(std::make_pair(province, newState.getID()));
 		if (auto coresMapping = coresMap.find(province); coresMapping != coresMap.end())
 		{
-			newState.addCores(coresMapping->second);
+			possibleCores.insert(coresMapping->second.begin(), coresMapping->second.end());
+		}
+	}
+
+	std::set<int> sourceProvinceNums;
+	for (const auto& province: provinceNumbers)
+	{
+		const auto possibleProvinces = provinceMapper.getHoI4ToVic2ProvinceMapping(province);
+		if (possibleProvinces)
+		{
+			sourceProvinceNums.insert(possibleProvinces->begin(), possibleProvinces->end());
+		}
+	}
+
+	for (const auto& [Vic2Core, HoI4Core]: possibleCores)
+	{
+		const auto sourceCountry = sourceCountries.find(Vic2Core);
+		if (sourceCountry == sourceCountries.end())
+		{
+			continue;
+		}
+		auto acceptedCultures = sourceCountry->second->getAcceptedCultures();
+		acceptedCultures.insert(sourceCountry->second->getPrimaryCulture());
+
+		uint64_t totalPopulation = 0;
+		double acceptedPopulation = 0;
+		for (const auto& sourceProvinceNum: sourceProvinceNums)
+		{
+			if (const auto& vic2Province = vic2Provinces.find(sourceProvinceNum); vic2Province != vic2Provinces.end())
+			{
+				const auto provincePopulation = vic2Province->second->getTotalPopulation();
+				totalPopulation += provincePopulation;
+				acceptedPopulation +=
+					 provincePopulation * vic2Province->second->getPercentageWithCultures(acceptedCultures);
+			}
+		}
+
+		if (acceptedPopulation / static_cast<double>(totalPopulation) >= 0.25)
```

Comment:
your accepted pops are a mix of primary pops + everything else accepted in those provinces. 0.25 seems low, to have a majority I'd shove this to 0.5 at minimum.